contracts/SCBridgeWallet.sol

```diff
@@ -194,6 +233,73 @@ contract SCBridgeWallet is IAccount {
   uint256 internal constant SIG_VALIDATION_SUCCEEDED = 0;
   uint256 internal constant SIG_VALIDATION_FAILED = 1;
 
+  /// This validates the crossChain UserOp
+  /// It ensures that it signed by the owner and intermediary on every chain
+  /// It also ensures that the UserOp targets the current chain and entrypoint
+  function validateCrossChain(UserOperation calldata userOp) private view {
+    (ExecuteChainInfo[] memory e, PaymentChainInfo[] memory p) = abi.decode(
+      userOp.callData[4:],
+      (ExecuteChainInfo[], PaymentChainInfo[])
+    );
+
+    bool foundExecute = false;
+    bool foundPayment = false;
+
+    // We expect every owner and intermediary on every chain to have signed the userOpHash
+    // For each chain we expect a signature from the owner and intermediary
+    require(
+      userOp.signature.length == (e.length + p.length) * 65 * 2,
+      "Invalid signature length"
+    );
+
+    for (uint i = 0; i < e.length; i++) {
+      if (e[i].chainId == block.chainid && e[i].entrypoint == entrypoint) {
+        foundExecute = true;
+      }
+
+      // TODO: I think we could just have everyone signed the UserOpHash for the first chain, instead of generating a new hash for each chain?
```

Comment:
Be good to resolve this before merging.

Reply:
[See https://github.com/magmo/Bridge-Wallet/pull/1/commits/c4829f5763967725576751055c721a6dcf98e949](https://github.com/magmo/Bridge-Wallet/pull/1/commits/c4829f5763967725576751055c721a6dcf98e949)